aquadoggo/src/db/stores/document.rs

```diff
@@ -366,76 +381,161 @@ impl SqlStore {
-            query(
+        Ok(document_view_ids
+            .iter()
+            .map(|document_id_str| {
+                document_id_str
+                    .parse::<DocumentViewId>()
+                    .expect("Document Id's coming from the store should be valid")
+            })
+            .collect())
+    }
+
+    /// Get the ids of all documents which are related to from another document view.
+    pub async fn get_child_document_ids(
+        &self,
+        document_view_id: &DocumentViewId,
+    ) -> Result<Vec<DocumentId>, DocumentStorageError> {
+        let document_view_ids: Vec<String> = query_scalar(&format!(
+            "
+            SELECT DISTINCT 
+                document_views.document_id
+            FROM
+                document_views
+            WHERE 
+                document_views.document_view_id 
+            IN (
+                SELECT
+                    operation_fields_v1.value
+                FROM 
+                    document_view_fields
+                LEFT JOIN 
+                    {OPERATION_FIELDS}
+                WHERE
+                    operation_fields_v1.field_type IN ('pinned_relation', 'pinned_relation_list')
+                AND 
+                    document_view_fields.document_view_id = $1
+            )
+            "
+        ))
+        .bind(document_view_id.to_string())
+        .fetch_all(&self.pool)
+        .await
+        .map_err(|err| DocumentStorageError::FatalStorageError(err.to_string()))?;
+
+        Ok(document_view_ids
+            .iter()
+            .map(|document_id_str| {
+                document_id_str
+                    .parse::<DocumentId>()
+                    .expect("Document Id's coming from the store should be valid")
+            })
+            .collect())
+    }
+
+    /// Attempt to remove a document view from the store. Returns a boolean which indicates if the
+    /// removal took place.
+    ///
+    /// This operations only succeeds if the view is "dangling", meaning no other document view
+    /// exists which relates to this view, AND it is not the current view of any document.
+    pub async fn prune_document_view(
+        &self,
+        document_view_id: &DocumentViewId,
+    ) -> Result<bool, DocumentStorageError> {
+        // Attempt to delete the view. If it is pinned from an existing view, or it is the current
+        // view of a document, the deletion will not go ahead.
+        let result = query(&format!(
                 "
                 DELETE FROM 
                     document_views
                 WHERE
                     document_views.document_view_id = $1
                 AND NOT EXISTS (
-                    SELECT * FROM operation_fields_v1
+                    SELECT 
+                        document_view_fields.document_view_id 
+                    FROM 
+                        document_view_fields
+                    LEFT JOIN
+                        {OPERATION_FIELDS}
```

Comment:
I would write the SQL query as-is in here, its harder to read with these constant values, same in all the other spots.